backend/oasst_backend/tree_manager.py

```diff
@@ -603,8 +605,34 @@ def check_condition_for_scoring_state(self, message_tree_id: UUID) -> bool:
                 return False
 
         self._enter_state(mts, message_tree_state.State.READY_FOR_SCORING)
+        self.update_message_ranks(rankings_by_message)
         return True
 
+    @managed_tx_method(CommitMode.COMMIT)
+    def update_message_ranks(self, rankings_by_message: Dict[int, int]) -> None:
+        for parent_msg_id, ranking in rankings_by_message.items():
+            unordered_ranking = []
+            for msg_reaction in ranking:
+                unordered_ranking.append(
+                    [msg_reaction.payload.payload.ranking, msg_reaction.payload.payload.ranked_message_ids]
+                )
+            sorted_ranking = self.sort_uuid_rankings(unordered_ranking)
+            sorted_ranking, sorted_uuids = zip(*sorted_ranking)
+            consensus = ranked_pairs(sorted_ranking)
+            for rank, uuid in zip(consensus, sorted_uuids[0]):
+                # set rank for each message_id for Message rows
+                msg = self.db.query(Message).filter(Message.id == uuid).one()
+                msg.rank = rank
+                self.db.add(msg)
+
+    def sort_uuid_rankings(self, data: List[Tuple[List[float], List[str]]]) -> List[Tuple[List[float], List[str]]]:
+        sorted_data = []
+        for rankings, uuids in data:
+            sorted_uuids = sorted(uuids)
```

Comment:
This looks a bit strange, e.g. the order of the uuids specifies the preference of the user .. if we sort it we destroy this information.